targets/TARGET_STM/can_api.c

```diff
@@ -190,19 +192,42 @@ int can_frequency(can_t *obj, int f)
     int pclk = HAL_RCC_GetPCLK1Freq();
     int btr = can_speed(pclk, (unsigned int)f, 1);
     CAN_TypeDef *can = (CAN_TypeDef *)(obj->can);
+    uint32_t tickstart = 0;
+    int status = 1;
 
     if (btr > 0) {
         can->MCR |= CAN_MCR_INRQ ;
+        /* Get tick */
+        tickstart = HAL_GetTick();
         while ((can->MSR & CAN_MSR_INAK) != CAN_MSR_INAK) {
+            if((HAL_GetTick() - tickstart ) > 2)
```

Comment:
Whitespace missing between `if` and `(`, also there's a redundant whitespace after `tickstart`.